cmd/nvidia-device-plugin/main.go

```diff
@@ -324,12 +338,7 @@ func startPlugins(c *cli.Context, flags []cli.Flag) ([]plugin.Interface, bool, e
 
 	// Get the set of plugins.
 	klog.Info("Retrieving plugins.")
-	kubeletSocket := c.String("kubelet-socket")
-	pluginManager, err := NewPluginManager(infolib, nvmllib, devicelib, kubeletSocket, config)
-	if err != nil {
-		return nil, false, fmt.Errorf("error creating plugin manager: %v", err)
-	}
-	plugins, err := pluginManager.GetPlugins()
+	plugins, err := GetPlugins(infolib, nvmllib, devicelib, config)
```

Comment:
I guess `kubeletSocket` isn't needed anymore to create a plugin?

Reply:
I don't feel too strongly, but my preference would be to keep it as a property of each plugin. I guess that is complicated by the fact that we don't instantiate each plugin from this package anymore (not that we would ever more than one path for this anyway).

Reply:
Like I said, no strong opinion though -- I can see the argument both ways...